### PR TITLE
no StatusError anymore, only NodeRuntimeError

### PR DESCRIPTION
**Pull request by nivanov-ati**

### CODE REVIEW

File: greenstalk.go [hunk 1]
```diff
@@ -42,22 +42,28 @@ func (bt *BehaviorTree[Blackboard]) Update(evt core.Event) core.Status {
 	result := core.Update(bt.ctx, bt.Root, bt.Blackboard, evt)
 
 	status := result.Status()
+	if status == core.StatusError {
+		if details, ok := result.(core.ErrorResultDetails); ok {
+			panic(details.Err)
+		} else {
+			// we currently only have NodeRuntimeError that return IsErroneous == true, so this
```
Comment (nivanov-ati) [inline greenstalk.go:49]: This comment is wrong now that the new changes have been applied

File: util/print_color.go [hunk 2]
```diff
@@ -39,8 +39,18 @@ func PrintTreeInColor[Blackboard any](node core.Node[Blackboard]) {
 
 func printInColor[Blackboard any](node core.Walkable[Blackboard], level int) {
 	indent := strings.Repeat("    ", level)
-	color.Set(colorForStatus[node.Status()])
-	fmt.Println(indent + node.String() + " " + symbolForStatus[node.Status()])
+
+	status := node.Result().Status()
```
Comment (nivanov-ati) [inline util/print_color.go:43]: I think this change can be undone

File: greenstalk.go (hunk 1 above)
Comment (jbcpollak) [reply greenstalk.go:49]: fixed

**Review by jbcpollak — APPROVED** [review]
LGTM

**Review by jbcpollak — APPROVED** [review]

**Pull request merged**